crates/paimon/src/spec/schema_change.rs

```diff
@@ -0,0 +1,736 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::spec::DataTypeRoot;
+use serde::{Deserialize, Serialize};
+
+/// Schema change to table.
+///
+/// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L36>
+#[derive(Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub enum SchemaChange {
```

Comment:
I think we don't need to specify types for every enum value. It's unlikely that we'll use `SetOption` without `SchemaChange`. 

How about embedding the struct directly in this enum?

```rust
pub enum SchemaChange {
    SetOption { 
        key: String,
        value: String,
    }
    ...
}
```